tests/regression/test_projection.py

```diff
@@ -249,28 +249,38 @@ def test_projector_bcs(tensor):
     if tensor == 'scalar':
         V = FunctionSpace(mesh, "CG", 1)
         V_ho = FunctionSpace(mesh, "CG", 5)
-        bc = DirichletBC(V_ho, Constant(42.0), (1, 2, 3, 4))
+        bcs = [DirichletBC(V_ho, Constant(0.5), (1, 3)),
+               DirichletBC(V_ho, Constant(-0.5), (2, 4))]
+        expr = Expression('cos(x[0]*pi*2)*sin(x[1]*pi*2)')
     elif tensor == 'vector':
         V = VectorFunctionSpace(mesh, "CG", 1)
         V_ho = VectorFunctionSpace(mesh, "CG", 5)
-        bc = DirichletBC(V_ho, Expression(("42.0", "42.0")),
-                         (1, 2, 3, 4))
+        bcs = [DirichletBC(V_ho, Expression(("0.5", "0.5")), (1, 3)),
+               DirichletBC(V_ho, Expression(("-0.5", "-0.5")), (2, 4))]
+        expr = Expression(['cos(x[0]*pi*2)*sin(x[1]*pi*2)']*2)
     elif tensor == 'tensor':
         V = TensorFunctionSpace(mesh, "CG", 1)
         V_ho = TensorFunctionSpace(mesh, "CG", 5)
-        bc = DirichletBC(V_ho, Expression(("42.0", "42.0",
-                                           "42.0", "42.0")),
-                         (1, 2, 3, 4))
-
-    exact = Function(V_ho)
-    bc.apply(exact)
+        bcs = [DirichletBC(V_ho, Expression(("0.5", "0.5",
+                                             "0.5", "0.5")), (1, 3)),
+               DirichletBC(V_ho, Expression(("-0.5", "-0.5",
+                                             "-0.5", "-0.5")), (2, 4))]
+        expr = Expression(['cos(x[0]*pi*2)*sin(x[1]*pi*2)']*4)
```

Comment:
I think it would be a bad idea to go around and replace existing uses of `Expression`s in the test suite, because then we would lose test coverage for that feature. However, I suggest to use UFL expressions in all new code, including test code, primarily because they are easier to get right because of the more helpful error messages.